src/main/java/com/github/dockerunit/discovery/consul/ConsulDescriptor.java

```diff
@@ -54,18 +54,13 @@ public CreateContainerCmd setup(CreateContainerCmd cmd) {
 
         return cmd.withHostConfig(hc)
                 .withExposedPorts(ports)
-                .withCmd("agent", "-dev", "-client=0.0.0.0", "-enable-script-checks");
+                .withCmd("sh", "-c", "consul agent -dev -client=0.0.0.0 -enable-script-checks -dns-port=53");
```

Comment:
If I run the agent directly, the official image is going to run as user `consul` and that prevents it from binding port 53.
The alternative would have been overwrite the rights that are assigned to that user, but it seemed quite dirty to me.
Also, I don't want to rely on a different image than the official one.

Reply:
Ah I see, you can maybe get around that by setting `CONSUL_ALLOW_PRIVILEGED_PORTS` ([link](https://github.com/hashicorp/docker-consul/blob/c5cfe2b2554acdbd06add9e71373b65d167b1ade/0.X/docker-entrypoint.sh#L91)) although it says it doesn't work with AUFS...

Anyway if the if shell is indeed the only option I think it would better to do `sh -c exec`, so we don't have a zombie/useless shell process running.

Reply:
Going with `sh -c exec`, thanks